### PR TITLE
Bug 2089563: Use PowerVS api types from openshift/api

### PR DESCRIPTION
**Pull request by Karthik-K-N**
Recently Power VS providerconfig has been added into openshift/api: https://github.com/openshift/api/pull/1159, This PR contains the changes to utiize those apis.

### CODE REVIEW

**Comment by Karthik-K-N** [comment]
/hold
Pending testing in Openshift cluster

File: cmd/manager/main.go [hunk 1]
```diff
@@ -139,31 +140,28 @@ func main() {
 	}
 
 	// Setup Scheme for all resources
-	if err := mapiv1beta1.AddToScheme(mgr.GetScheme()); err != nil {
+	if err := machinev1beta1.Install(mgr.GetScheme()); err != nil {
```
Comment (mkumatag) [inline cmd/manager/main.go:143]: why not `machinev1.Install`?

File: pkg/actuators/machine/actuator_test.go [hunk 2]
```diff
@@ -234,6 +233,9 @@ func TestActuatorEvents(t *testing.T) {
 				return nil
 			}
 			gs.Eventually(waitForEvent, timeout).Should(Succeed())
+			fmt.Println(eventList.Items[0].Message)
+			fmt.Println()
+			fmt.Println(tc.event)
```
Comment (mkumatag) [inline pkg/actuators/machine/actuator_test.go:238]: why these additional printfs?

File: pkg/actuators/machine/instances_test.go [hunk 3]
```diff
@@ -151,29 +151,31 @@ func TestGetServiceInstanceID(t *testing.T) {
...
+				Type: machinev1.PowerVSResourceTypeName,
 				Name: core.StringPtr(instanceName),
 				ID:   core.StringPtr(instanceID),
 			},
 			getCloudServiceInstancesValues: []bluemixmodels.ServiceInstanceV2{
 				{
 					ServiceInstance: bluemixmodels.ServiceInstance{
 						MetadataType: &bluemixmodels.MetadataType{
-							ID: instanceID,
+							Guid: instanceID,
```
Comment (mkumatag) [inline pkg/actuators/machine/instances_test.go:178]: why this flip? `ID` => `Guid`?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cmd/manager/main.go (hunk 1 above)
Comment (Karthik-K-N) [reply cmd/manager/main.go:143]: I have renamed the import from `mapiv1beta1 "github.com/openshift/api/machine/v1beta1"` to `machinev1beta1 "github.com/openshift/api/machine/v1beta1"` so added `machinev1.Install` at line no: 147

File: pkg/actuators/machine/actuator_test.go (hunk 2 above)
Comment (Karthik-K-N) [reply pkg/actuators/machine/actuator_test.go:238]: By mistake will remove

File: pkg/actuators/machine/instances_test.go (hunk 3 above)
Comment (Karthik-K-N) [reply pkg/actuators/machine/instances_test.go:178]: Guid is the one which actually containst the serviceInstance id

File: cmd/manager/main.go (hunk 1 above)
Comment (Karthik-K-N) [reply cmd/manager/main.go:143]: My thinking was v1 contains Power VS Provider conf and v1beta1 contains machine and machineset crds. so i added both but i think we need to install v1beta1 for registering machine and machineset crds and avoid installing v1, Need to check

Comment (Karthik-K-N) [reply cmd/manager/main.go:142]: removed machinev1.Install in the latest commit

**Comment by Karthik-K-N** [comment]
As we got the approval for the installer [PR](https://github.com/openshift/installer/pull/5873), We are almost good to merge this
/hold cancel

**Comment by Karthik-K-N** [comment]
@mkumatag can this be merged

File: pkg/actuators/machine/machine_scope.go [hunk 4]
```diff
@@ -5,15 +5,17 @@ import (
 	"fmt"
 	"strings"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (mkumatag) [inline pkg/actuators/machine/machine_scope.go:8]: this import can be clubbed with the below k8s block of imports

**Review by mkumatag — CHANGES_REQUESTED** [review]
few minor comments

File: pkg/actuators/machine/machine_scope.go (hunk 4 above)
Comment (Karthik-K-N) [reply pkg/actuators/machine/machine_scope.go]: Updated it

File: pkg/actuators/machine/actuator_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Karthik-K-N) [reply pkg/actuators/machine/actuator_test.go]: Removed

File: examples/machine-with-user-data.yaml [hunk 5]
```diff
@@ -2,27 +2,30 @@
 apiVersion: machine.openshift.io/v1beta1
 kind: Machine
 metadata:
-  generateName: powervs-actuator-testing-machine-
+  generateName: powervs-actuator-testing-machine-1
```
[the remark this answers is not in the document]
Comment (Karthik-K-N) [reply examples/machine-with-user-data.yaml]: Unintentional, Removed

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**